onnxruntime/core/optimizer/initializer.h

```diff
@@ -26,7 +26,9 @@ class Initializer final {
     size_ = std::accumulate(dims_.begin(), dims_.end(), int64_t(1), std::multiplies<int64_t>{});
 
     switch (data_type_) {
-      case ONNX_NAMESPACE::TensorProto_DataType_FLOAT16: {
+      case ONNX_NAMESPACE::TensorProto_DataType_FLOAT16:
+      //reuse float16 field
+      case ONNX_NAMESPACE::TensorProto_DataType_BFLOAT16: {
```

Comment:
math::floatToHalf is for fp16. Seems need a new case for bf16. Or are they same for zero value?

      case ONNX_NAMESPACE::TensorProto_DataType_BFLOAT16: {
        // Reuse float16 field
        float16_data_.assign(static_cast<size_t>(size_), BFloat16(0.f).val);
        break;
      }

Reply:
it is initialize to 0 so either is fine. but u r right, it is better to make it explicit.